File: test-suite-docs-java/src/test/java/io/micronaut/discovery/docs/HelloWorldServiceConfigurer.java
package io.micronaut.discovery.docs;

import io.micronaut.context.ApplicationContextBuilder;
import io.micronaut.context.ApplicationContextConfigurer;
import io.micronaut.context.annotation.ContextConfigurer;
import io.micronaut.core.io.socket.SocketUtils;

import java.util.Map;

/**
 * Binds the embedded server to a free port and registers that port as the fixed URL of the
 * {@code hello-world} service, so the examples resolve the service without a Consul agent.
 */
@ContextConfigurer
public class HelloWorldServiceConfigurer implements ApplicationContextConfigurer {

    @Override
    public void configure(ApplicationContextBuilder builder) {
        int port = SocketUtils.findAvailableTcpPort();
        builder.properties(Map.of(
                "micronaut.server.port", port,
                "micronaut.http.services.hello-world.url", "http://localhost:" + port));
    }
}

File: test-suite-docs-kotlin/src/test/kotlin/io/micronaut/discovery/docs/HelloWorldServiceConfigurer.kt
package io.micronaut.discovery.docs

import io.micronaut.context.ApplicationContextBuilder
import io.micronaut.context.ApplicationContextConfigurer
import io.micronaut.context.annotation.ContextConfigurer
import io.micronaut.core.io.socket.SocketUtils

/**
 * Binds the embedded server to a free port and registers that port as the fixed URL of the
 * `hello-world` service, so the examples resolve the service without a Consul agent.
 */
@ContextConfigurer
class HelloWorldServiceConfigurer : ApplicationContextConfigurer {

    override fun configure(builder: ApplicationContextBuilder) {
        val port = SocketUtils.findAvailableTcpPort()
        builder.properties(mapOf(
            "micronaut.server.port" to port,
            "micronaut.http.services.hello-world.url" to "http://localhost:$port"))
    }
}
